app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,27 @@
+class SalesController < ApplicationController
+  def new
+    @vendor = Vendor.find(params[:vendor_id])
+    @sale = Sale.new
+  end
+
+  def create
+    @vendor = Vendor.find(params[:vendor_id])
+    @sale = Sale.new(create_sale_params[:sale])
+    if @sale.save
+      # We thought the purchase time should be generated at the time the Sale is created
+      # as if from a POS system, we didn't think Vendor should have to manually input
+      # the date and time of the sale
+      @sale.update(vendor_id: params[:vendor_id], purchase_time: Time.now, product_id: create_sale_params[:sale][:product_id])
+      # raise
```

Comment:
I love that you were testing this out by putting a raise here! Not a big deal, but it's good to go through commented out code and clean it up at the end.